demos/demo03_poisson_circular.py

```diff
@@ -63,7 +64,7 @@ def u_boundary(x, on_boundary):
 
 # 2.1.2 The Matern fieled (maps i.i.d normal random vector of dimension `num_terms`
 # to Matern field realization on `fenics_continuous_geo`)
-matern_geo = cuqipy_fenics.geometry.MaternKLExpansion(fenics_continuous_geo, length_scale = .8, num_terms=128)
+matern_geo = cuqipy_fenics.geometry.MaternKLExpansion(fenics_continuous_geo, length_scale = .8, num_terms=12)
```

Comment:
Why the change in parameter value? That seems unrelated to the issue/PR scope of updating to use new sampler?